### ISSUE TITLE
Use `toLowerCase(Locale.ROOT)` instead of `toLowerCase()`

### ISSUE THREAD

**Issue by AterAnimAvis** [Alex-the-666/Ice_and_Fire#3345]
### Explanation
You are currently calling `toLowerCase` without providing a `Locale` on Enum names to generate `ResourceLocation`, this works for most usesbut is actually locale dependent.

Java's `toLowerCase` is by default Language Dependent (Unless you pass a `Locale` in). Something is creating a RL with toLowerCase without a Locale, meaning it uses the system locale. For most locales this doesn't matter, Turkish however does cause issues.

From the Javadocs.

	For instance, "TITLE".toLowerCase() in a Turkish locale returns "tıtle", where 'ı' is the LATIN SMALL LETTER DOTLESS I character.

### Replicating

You can pass in `-Duser.language=tr -Duser.country=TR` as JVM Parameters to replicate having Turkish as your default locale. 

### Example Crash 

from `iceandfire-2.1.4-1.16.4.jar`
```
java.lang.ExceptionInInitializerError: null
	at java.lang.Class.forName0(Native Method) ~[?:1.8.0_51]
	at java.lang.Class.forName(Class.java:348) ~[?:1.8.0_51]
	at net.minecraftforge.fml.AutomaticEventSubscriber.lambda$inject$6(AutomaticEventSubscriber.java:75) ~[?:?]
	at net.minecraftforge.fml.AutomaticEventSubscriber$$Lambda$3150/1571354076.accept(Unknown Source) ~[?:?]
	at java.util.ArrayList.forEach(ArrayList.java:1249) ~[?:1.8.0_51]
	at net.minecraftforge.fml.AutomaticEventSubscriber.inject(AutomaticEventSubscriber.java:62) ~[?:?]
	at net.minecraftforge.fml.javafmlmod.FMLModContainer.constructMod(FMLModContainer.java:91) [?:35.1]
	at net.minecraftforge.fml.javafmlmod.FMLModContainer$$Lambda$3080/175822565.run(Unknown Source) [?:35.1]
	at net.minecraftforge.fml.ModContainer.lambda$buildTransitionHandler$4(ModContainer.java:120) [?:?]
	at net.minecraftforge.fml.ModContainer$$Lambda$3112/1640832113.run(Unknown Source) [?:?]
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1618) [?:1.8.0_51]
	at java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1610) [?:1.8.0_51]
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289) [?:1.8.0_51]
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056) [?:1.8.0_51]
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1689) [?:1.8.0_51]
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:157) [?:1.8.0_51]
Caused by: net.minecraft.util.ResourceLocationException: Non [a-z0-9/._-] character in path of location: iceandfire:h?ppogryph_skull
	at net.minecraft.util.ResourceLocation.<init>(ResourceLocation.java:33) ~[?:?]
	at net.minecraft.util.ResourceLocation.<init>(ResourceLocation.java:42) ~[?:?]
	at net.minecraftforge.registries.GameData.checkPrefix(GameData.java:898) ~[?:?]
	at net.minecraftforge.registries.ForgeRegistryEntry.checkRegistryName(ForgeRegistryEntry.java:71) ~[?:?]
	at net.minecraftforge.registries.ForgeRegistryEntry.setRegistryName(ForgeRegistryEntry.java:45) ~[?:?]
	at net.minecraftforge.registries.ForgeRegistryEntry.setRegistryName(ForgeRegistryEntry.java:52) ~[?:?]
	at com.github.alexthe666.iceandfire.item.ItemMobSkull.<init>(ItemMobSkull.java:22) ~[?:2.1.4-1.16.4]
	at com.github.alexthe666.iceandfire.enums.EnumSkullType.initItems(EnumSkullType.java:26) ~[?:2.1.4-1.16.4]
	at com.github.alexthe666.iceandfire.item.IafItemRegistry.<clinit>(IafItemRegistry.java:333) ~[?:2.1.4-1.16.4]
	... 16 more
```

### TLDR: 
Pass `Locale.ROOT` to `toLowerCase` to prevent `ResourceLocation` issues when the Default Locale is Turkish.

This was originally reported to the ForgeDiscord [here](https://discord.com/channels/313125603924639766/437001959950778368/790234155086249994).
Passing this on as I can't find a report being submitted (I replicated the issue using the VM parameters above, removing the parameters so it uses my default locale of en_US fixed the issue).

(Note that LATIN SMALL LETTER DOTLESS I may display as ? on some systems)
